internal/cache/memcache/memory_cache_test.go

```diff
@@ -0,0 +1,83 @@
+package memcache_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/RHEnVision/provisioning-backend/internal/cache/memcache"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSetAndGetFound(t *testing.T) {
+	c := memcache.New[string, string](0)
+	c.Set("hello", "Hello", 0)
+	hello, found := c.Get("hello")
+	assert.True(t, found)
+	assert.Equal(t, "Hello", hello)
+}
+
+func TestSetAndGetNotFound(t *testing.T) {
+	c := memcache.New[string, string](0)
+	_, found := c.Get("does not exist")
+	assert.False(t, found)
+}
+
+func TestManualExpiration(t *testing.T) {
+	c := memcache.New[string, string](time.Minute)
+	c.Set("short", "expiration", time.Nanosecond)
+	c.ExpireNow()
+
+	_, found := c.Get("short")
+	assert.False(t, found)
+}
+
+func TestExpiration(t *testing.T) {
+	c := memcache.New[string, string](10 * time.Millisecond)
+	c.Set("short", "expiration", time.Nanosecond)
+	defer c.Stop()
+
+	// hope for the best
+	time.Sleep(100 * time.Millisecond)
+
+	_, found := c.Get("short")
+	assert.False(t, found)
+}
```

Comment:
This test is clunky, however, I think we can keep it. If this would fail (e.g. on CI), let's remove it. It is almost all covered in the `TestManualExpiration` above (except it's not triggered via the timer).